src/test/java/lambda/part1/exercise/Exercise2.java

```diff
@@ -0,0 +1,123 @@
+package lambda.part1.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.Function;
+
+import static java.util.stream.Collectors.averagingDouble;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+class Exercise2 {
+
+    @Test
+    void calcAverageAgeOfEmployees() {
+        List<Employee> employees = getEmployees();
+
+        Double expected = employees.stream()
+                .map(Employee::getPerson)
+                .collect(averagingDouble(Person::getAge));
+
+        assertThat(expected, Matchers.closeTo(33.66, 0.1));
+    }
+
+    @Test
+    void findPersonWithLongestFullName() {
+        List<Employee> employees = getEmployees();
+
+        Person expected = employees.stream()
+                .map(Employee::getPerson)
+                .max(Comparator.comparingInt((Person p) -> p.getFullName().length()))
+                .orElse(employees.get(0).getPerson());
+
+        assertThat(expected, Matchers.is(employees.get(1).getPerson()));
+    }
+
+    @Test
+    void findEmployeeWithMaximumDurationAtOnePosition() {
+        List<Employee> employees = getEmployees();
+
+        Function<Employee, Integer> employeeToMaxDuration = (Employee employee) -> employee.getJobHistory()
+                .stream()
+                .map(JobHistoryEntry::getDuration)
+                .max(Comparator.comparingInt(Integer::intValue))
+                .orElse(0);
+
+        Employee expected = employees.stream()
+                .max(Comparator.comparing(employeeToMaxDuration))
+                .orElse(employees.get(0));
+
+        assertThat(expected, Matchers.is(employees.get(4)));
+    }
+
+    /**
+     * Вычислить общую сумму заработной платы для сотрудников.
+     * Базовая ставка каждого сотрудника составляет 75_000.
+     * Если на текущей позиции (последняя в списке) он работает больше трех лет - ставка увеличивается на 20%
+     */
+    @Test
+    void calcTotalSalaryWithCoefficientWorkExperience() {
+        List<Employee> employees = getEmployees();
+
+        Double expected = employees.stream()
+                .map(Employee::getJobHistory)
+                .map(jobs -> jobs.get(jobs.size() - 1))
+                .map(JobHistoryEntry::getDuration)
+                .map(duration -> duration > 3 ? 75000d : 75000 * 1.2)
```

Comment:
Did you run test for this? it should we vice versa

Reply:
i actually thought the value in test is incorrect, so i changed it, am gonna ask about it on lecture

Reply:
oh, i see my mistake now